From 876f9f4a27ffd9477be8348710cb0baf387c50a0 Mon Sep 17 00:00:00 2001
From: Leah Neukirchen <leah@vuxu.org>
Date: Mon, 7 Aug 2017 16:08:15 +0200
Subject: [PATCH] xbps-fbulk: use _DEFAULT_SOURCE, required for wait3 in glibc
 2.16.

Also these guards need to be defined before including anything.
---
 bin/xbps-fbulk/main.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/bin/xbps-fbulk/main.c b/bin/xbps-fbulk/main.c
index 82dc0a16..b686cfa7 100644
--- bin/xbps-fbulk/main.c
+++ bin/xbps-fbulk/main.c
@@ -47,6 +47,8 @@
  * Only one attempt is made to build any given package, no matter how many
  * other packages depend on it.
  */
+#define _DEFAULT_SOURCE
+#define _BSD_SOURCE
 #include <stdio.h>
 #include <stdlib.h>
 #include <unistd.h>
@@ -59,9 +61,7 @@
 #include <fcntl.h>
 #include <sys/file.h>
 #include <sys/stat.h>
-#define _BSD_SOURCE
-# include <sys/wait.h>
-#undef _BSD_SOURCE
+#include <sys/wait.h>
 #include <getopt.h>
 
 #include <xbps.h>
